src/libostree/ostree-checksum-input-stream.c

```diff
@@ -132,7 +132,11 @@ ostree_checksum_input_stream_read (GInputStream *stream, void *buffer, gsize cou
 
   res = g_input_stream_read (fself->base_stream, buffer, count, cancellable, error);
   if (res > 0)
-    g_checksum_update (self->priv->checksum, buffer, res);
+    {
+      guchar *char_buffer = (guchar *)buffer;
+      char_buffer[res] = '\0';
+      g_checksum_update (self->priv->checksum, (const guchar *)char_buffer, res);
```

Comment:
This can't be right. Data is being read from a stream to a buffer and being checksummed. We can't just arbitrarily add a NUL to it and expect to get the right checksum. I don't know why coverity thinks this is a string that needs terminating.